stl/inc/yvals_core.h

```diff
@@ -1327,7 +1326,15 @@
 #if _HAS_CXX20
 #define __cpp_lib_array_constexpr 201811L // P1032R1 Miscellaneous constexpr
 #elif _HAS_CXX17 // ^^^ _HAS_CXX20 / _HAS_CXX17 vvv
-#define __cpp_lib_array_constexpr 201803L
+#define __cpp_lib_array_constexpr 201803L // P0858R0 Constexpr Iterator Requirements
+#endif // _HAS_CXX17
+
+#if _HAS_CXX20 && (defined(__clang__) || defined(__EDG__)) // TRANSITION, DevCom-1331017

```

Comment:
I won't nitpick location since this entire "C++20 with C++17 fallback" section is a mess, but we really should reorder all of 1318-1356 properly in a followup.

Reply:
Could we just disable C4005 around all of the feature test macros and simply overwrite these macros? I think it would be much simpler.
```
#if _HAS_CXX17
// ...
#define __cpp_lib_optional 201606L
// ...
#endif // _HAS_CXX17

#if _HAS_CXX20
// ...
#if defined(__clang__) || defined(__EDG__)
#define __cpp_lib_optional 202106L
#endif
// ...
#endif // _HAS_CXX20
```

Reply:
Filed this discussion as https://github.com/microsoft/STL/issues/2135.